test/unit-test/FreeRTOS_UDP_IP/FreeRTOS_UDP_IP_utest.c

```diff
@@ -74,44 +72,25 @@
     #define ipconfigTCP_MSS    ( ipconfigNETWORK_MTU - ( ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_TCP_HEADER ) )
 #endif
 
+void vProcessGeneratedUDPPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer );
+BaseType_t xProcessReceivedUDPPacket( NetworkBufferDescriptor_t * pxNetworkBuffer,
+                                      uint16_t usPort,
+                                      BaseType_t * pxIsWaitingForARPResolution );
 
-extern NetworkInterface_t xInterfaces[ 1 ];
-
-static uint32_t ulFunctionCalled = 0;
-static BaseType_t xFunctionReturn;
-static BaseType_t xLocalHandler( Socket_t pxSocket,
-                                 void * pvData,
-                                 size_t xLength,
-                                 const struct freertos_sockaddr * pxFrom,
-                                 const struct freertos_sockaddr * pxTo )
-{
-    TEST_ASSERT( pxSocket != NULL );
-    TEST_ASSERT( pvData != NULL );
-    TEST_ASSERT( pxFrom != NULL );
-    TEST_ASSERT( pxTo != NULL );
-
-    ulFunctionCalled++;
+/* ==============================  Test Cases  ============================== */
 
-    return xFunctionReturn;
-}
-
-
-static void vConfigureInterfaceAndEndpoints( NetworkBufferDescriptor_t * xLocalNetworkBuffer,
-                                             struct xNetworkEndPoint * xEndPoint,
-                                             struct xNetworkInterface * xInterface )
+/**
+ * @brief Input with null network buffer pointer.
+ */
+void test_vProcessGeneratedUDPPacket_NullNetworkBuffer( void )
 {
-    xEndPoint->pxNetworkInterface = xInterface;
-    xEndPoint->pxNext = NULL;
-    xLocalNetworkBuffer->pxEndPoint = xEndPoint;
-    xInterface->pxEndPoint = xEndPoint;
-    xInterface->pxNext = NULL;
+    vProcessGeneratedUDPPacket( NULL );
 }
 
-/*
- * @brief Test what happens if the packet cannot be sent due to
- *        the address not being resolved.
+/**
+ * @brief Pass IPv6 packet to IPv4 API to handle.
```

Comment:
Comment should be : Pass IPv4 packet to IPv4 API to handle

Reply:
Updated